galsim/chromatic.py

```diff
@@ -265,11 +319,7 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
         wave_list = self._getCombinedWaveList(bandpass)
 
         if self.separable:
-            if len(wave_list) > 0:
-                multiplier = np.trapz(self.SED(wave_list) * bandpass(wave_list), wave_list)
-            else:
-                multiplier = galsim.integ.int1d(lambda w: self.SED(w) * bandpass(w),
-                                                bandpass.blue_limit, bandpass.red_limit)
+            multiplier = _multiplier_cache(self.SED, bandpass, tuple(wave_list))
```

Comment:
Why do you call tuple here?  Does a numpy array not work as a key for the cache for some reason?